In [30]:
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [31]:
%sql mysql+mysqldb://USERNAME:PASSWORD@HOST/DATABASE

'Connected: admin@sql_project'

# 5 SQL Queries # 


### 1. Find video games with no critic score  ### 

In [39]:
%%sql
SELECT *
FROM GameScore 
WHERE CriticScore = 0;

 * mysql+mysqldb://admin:***@lmu-dev-01.c205jaszqskd.us-east-1.rds.amazonaws.com/sql_project
   mysql://admin:***@lmu-dev-01.c205jaszqskd.us-east-1.rds.amazonaws.com/sql_project
30 rows affected.


[(6, 'The Witcher 3: Wild Hunt    ', 'Warner Bros. Interactive Entertainment  ', Decimal('0.0')),
 (7, 'Call of Duty: Black Ops IIII    ', 'Activision  ', Decimal('0.0')),
 (8, 'Call of Duty: Infinite Warfare    ', 'Activision  ', Decimal('0.0')),
 (13, 'Uncharted: The Nathan Drake Collection    ', 'Sony Computer Entertainment  ', Decimal('0.0')),
 (14, 'Final Fantasy VII Remake    ', 'Square Enix  ', Decimal('0.0')),
 (16, "Tom Clancy's The Division    ", 'Ubisoft  ', Decimal('0.0')),
 (17, 'Watch Dogs    ', 'Ubisoft  ', Decimal('0.0')),
 (18, "Assassin's Creed: Unity    ", 'Ubisoft  ', Decimal('0.0')),
 (25, 'Mortal Kombat X    ', 'Warner Bros. Interactive Entertainment  ', Decimal('0.0')),
 (27, 'Call of Duty: Black Ops 3    ', 'Activision  ', Decimal('0.0')),
 (32, 'FIFA 19    ', 'Electronic Arts  ', Decimal('0.0')),
 (37, 'Destiny    ', 'Activision  ', Decimal('0.0')),
 (39, 'Crash Bandicoot N. Sane Trilogy    ', 'Activision  ', Decimal('0.0')),
 (40, 'Star Wars Battlefront II (2017)    ', 'Electronic Arts  ', Decimal('0.0')),
 (41, "Tom Clancy's Rainbow Six: Siege    ", 'Ubisoft  ', Decimal('0.0')),
 (43, 'Destiny 2    ', 'Activision  ', Decimal('0.0')),
 (44, "Assassin's Creed Origins    ", 'Ubisoft  ', Decimal('0.0')),
 (45, 'The Last of Us Part II    ', 'Sony Interactive Entertainment  ', Decimal('0.0')),
 (48, 'Gran Turismo Sport    ', 'Sony Interactive Entertainment  ', Decimal('0.0')),
 (50, 'FIFA 14    ', 'EA Sports  ', Decimal('0.0')),
 (57, 'FIFA 19    ', 'Electronic Arts  ', Decimal('0.0')),
 (62, 'Destiny    ', 'Activision  ', Decimal('0.0')),
 (64, 'Crash Bandicoot N. Sane Trilogy    ', 'Activision  ', Decimal('0.0')),
 (65, 'Star Wars Battlefront II (2017)    ', 'Electronic Arts  ', Decimal('0.0')),
 (66, "Tom Clancy's Rainbow Six: Siege    ", 'Ubisoft  ', Decimal('0.0')),
 (68, 'Destiny 2    ', 'Activision  ', Decimal('0.0')),
 (69, "Assassin's Creed Origins    ", 'Ubisoft  ', Decimal('0.0')),
 (70, 'The Last of Us Part II    ', 'Sony Interactive Entertainment  ', Decimal('0.0')),
 (73, 'Gran Turismo Sport    ', 'Sony Interactive Entertainment  ', Decimal('0.0')),
 (75, 'FIFA 14    ', 'EA Sports  ', Decimal('0.0'))]

There are 30/75 games that do not have a critic score

### 2. Find games with more than 1.5million sold in NA ###

In [41]:
%%sql
SELECT 
    DISTINCT(Game), 
    NASales,
    Releaseyear
FROM GameSales 
WHERE NASales > 1.5
ORDER BY NASales DESC;

 * mysql+mysqldb://admin:***@lmu-dev-01.c205jaszqskd.us-east-1.rds.amazonaws.com/sql_project
   mysql://admin:***@lmu-dev-01.c205jaszqskd.us-east-1.rds.amazonaws.com/sql_project
31 rows affected.


[('Call of Duty: Black Ops 3', Decimal('6.18'), 2015),
 ('Grand Theft Auto V', Decimal('6.06'), 2014),
 ('Red Dead Redemption 2', Decimal('5.26'), 2018),
 ('Call of Duty: WWII', Decimal('4.67'), 2017),
 ('Uncharted (PS4)', Decimal('4.49'), 2016),
 ('Spider-Man (PS4)', Decimal('3.64'), 2018),
 ('Star Wars Battlefront 2015', Decimal('3.31'), 2015),
 ('Call of Duty: Infinite Warfare', Decimal('3.11'), 2016),
 ('Fallout 4', Decimal('2.91'), 2015),
 ('Call of Duty: Advanced Warfare', Decimal('2.84'), 2014),
 ('God of War (PS4)', Decimal('2.83'), 2018),
 ('The Last of Us', Decimal('2.70'), 2014),
 ('NBA 2K16', Decimal('2.56'), 2015),
 ('Uncharted: The Nathan Drake Collection', Decimal('2.55'), 2015),
 ('Destiny', Decimal('2.53'), 2014),
 ('Madden NFL 16', Decimal('2.39'), 2015),
 ('Madden NFL 17', Decimal('2.30'), 2016),
 ('NBA 2K17', Decimal('2.28'), 2016),
 ('Battlefield 1', Decimal('2.20'), 2016),
 ('Horizon: Zero Dawn', Decimal('2.20'), 2017),
 ('NBA 2K18', Decimal('2.13'), 2017),
 ('Mortal Kombat X', Decimal('1.94'), 2015),
 ('Destiny 2', Decimal('1.92'), 2017),
 ('MineCraft', Decimal('1.89'), 2014),
 ('Overwatch', Decimal('1.84'), 2016),
 ('Final Fantasy XV', Decimal('1.81'), 2016),
 ('Call of Duty: Ghosts', Decimal('1.79'), 2013),
 ('Madden NFL 18', Decimal('1.73'), 2017),
 ('Star Wars Battlefront II (2017)', Decimal('1.70'), 2017),
 ('Batman: Arkham Knight', Decimal('1.65'), 2015),
 ('Madden NFL 15', Decimal('1.58'), 2014)]

31 Games from 2014-2018 have sold more than 1.5 million copies.

### 3. Find duplicate values ###

In [46]:
%%sql
SELECT 
    Game, 
    COUNT(*)
FROM GameSales 
GROUP BY Game 
HAVING COUNT(*) > 1;

 * mysql+mysqldb://admin:***@lmu-dev-01.c205jaszqskd.us-east-1.rds.amazonaws.com/sql_project
   mysql://admin:***@lmu-dev-01.c205jaszqskd.us-east-1.rds.amazonaws.com/sql_project
1 rows affected.


[('Grand Theft Auto V', 2)]

There are duplicate game titles, use DISTINCT

### 4. List the Publishers and number of games for each ###

In [ ]:
%%sql 
SELECT 
    Publisher,
    COUNT(DISTINCT(Game)) AS NumOfGames
FROM GameSales
GROUP BY Publisher
ORDER BY COUNT(DISTINCT(Game)) DESC;

Lists all game publishers and amount of games under each publisher.

### 5. Identify the range of release years in the data ###

In [ ]:
%%sql
SELECT 
    MAX(ReleaseYear),
    MIN(ReleaseYear)
FROM GameSales;

Lists games from 2013 to 2018. Games released in 2018 will continue to finish selling while older games have reached their max.

# Games with the highest sales in North America > than the average #

### Primary Question: Which PS4 video games and their genres are the highest sellers in North America? ###

Getting the above average sellers and genre to see what titles have the highest sales.

CTE, Subquery, JOIN

In [36]:
%%sql
WITH GameSalesCTE AS (
    SELECT 
        DISTINCT(Game), 
        UCASE(GenreName),
        NASales
    FROM GameSales s
    JOIN Genre g
        ON s.GenreID = g.GenreID
    ORDER BY NASales DESC
)
SELECT *,
    (
        SELECT AVG(NASales)
        FROM GameSalesCTE
    ) AS AvgNASales
FROM GameSalesCTE
WHERE NASales > (
    SELECT AVG(NASales)
    FROM GameSalesCTE
);

 * mysql+mysqldb://admin:***@lmu-dev-01.c205jaszqskd.us-east-1.rds.amazonaws.com/sql_project
   mysql://admin:***@lmu-dev-01.c205jaszqskd.us-east-1.rds.amazonaws.com/sql_project
102 rows affected.


[('Call of Duty: Black Ops 3', 'SHOOTER', Decimal('6.18'), Decimal('0.563464')),
 ('Grand Theft Auto V', 'ACTION', Decimal('6.06'), Decimal('0.563464')),
 ('Red Dead Redemption 2', 'ACTION', Decimal('5.26'), Decimal('0.563464')),
 ('Call of Duty: WWII', 'SHOOTER', Decimal('4.67'), Decimal('0.563464')),
 ('Uncharted (PS4)', 'ACTION', Decimal('4.49'), Decimal('0.563464')),
 ('Spider-Man (PS4)', 'ACTION', Decimal('3.64'), Decimal('0.563464')),
 ('Star Wars Battlefront 2015', 'SHOOTER', Decimal('3.31'), Decimal('0.563464')),
 ('Call of Duty: Infinite Warfare', 'SHOOTER', Decimal('3.11'), Decimal('0.563464')),
 ('Fallout 4', 'ROLE-PLAYING', Decimal('2.91'), Decimal('0.563464')),
 ('Call of Duty: Advanced Warfare', 'SHOOTER', Decimal('2.84'), Decimal('0.563464')),
 ('God of War (PS4)', 'ACTION', Decimal('2.83'), Decimal('0.563464')),
 ('The Last of Us', 'ACTION', Decimal('2.70'), Decimal('0.563464')),
 ('NBA 2K16', 'SPORTS', Decimal('2.56'), Decimal('0.563464')),
 ('Uncharted: The Nathan Drake Collection', 'ACTION', Decimal('2.55'), Decimal('0.563464')),
 ('Destiny', 'SHOOTER', Decimal('2.53'), Decimal('0.563464')),
 ('Madden NFL 16', 'SPORTS', Decimal('2.39'), Decimal('0.563464')),
 ('Madden NFL 17', 'SPORTS', Decimal('2.30'), Decimal('0.563464')),
 ('NBA 2K17', 'SPORTS', Decimal('2.28'), Decimal('0.563464')),
 ('Horizon: Zero Dawn', 'ACTION', Decimal('2.20'), Decimal('0.563464')),
 ('Battlefield 1', 'SHOOTER', Decimal('2.20'), Decimal('0.563464')),
 ('NBA 2K18', 'SPORTS', Decimal('2.13'), Decimal('0.563464')),
 ('Mortal Kombat X', 'FIGHTING', Decimal('1.94'), Decimal('0.563464')),
 ('Destiny 2', 'SHOOTER', Decimal('1.92'), Decimal('0.563464')),
 ('MineCraft', 'MISC', Decimal('1.89'), Decimal('0.563464')),
 ('Overwatch', 'SHOOTER', Decimal('1.84'), Decimal('0.563464')),
 ('Final Fantasy XV', 'ROLE-PLAYING', Decimal('1.81'), Decimal('0.563464')),
 ('Call of Duty: Ghosts', 'SHOOTER', Decimal('1.79'), Decimal('0.563464')),
 ('Madden NFL 18', 'SPORTS', Decimal('1.73'), Decimal('0.563464')),
 ('Star Wars Battlefront II (2017)', 'SHOOTER', Decimal('1.70'), Decimal('0.563464')),
 ('Batman: Arkham Knight', 'ACTION', Decimal('1.65'), Decimal('0.563464')),
 ('Madden NFL 15', 'SPORTS', Decimal('1.58'), Decimal('0.563464')),
 ("Tom Clancy's The Division", 'SHOOTER', Decimal('1.49'), Decimal('0.563464')),
 ('NBA 2K15', 'SPORTS', Decimal('1.49'), Decimal('0.563464')),
 ('The Witcher 3: Wild Hunt', 'ROLE-PLAYING', Decimal('1.48'), Decimal('0.563464')),
 ('Far Cry 5', 'ACTION', Decimal('1.44'), Decimal('0.563464')),
 ('Watch Dogs', 'ACTION', Decimal('1.40'), Decimal('0.563464')),
 ('Battlefield 4', 'SHOOTER', Decimal('1.40'), Decimal('0.563464')),
 ('inFAMOUS: Second Son', 'ACTION', Decimal('1.35'), Decimal('0.563464')),
 ('Bloodborne', 'ROLE-PLAYING', Decimal('1.30'), Decimal('0.563464')),
 ("Tom Clancy's Rainbow Six: Siege", 'SHOOTER', Decimal('1.29'), Decimal('0.563464')),
 ('FIFA 18', 'SPORTS', Decimal('1.27'), Decimal('0.563464')),
 ('FIFA 17', 'SPORTS', Decimal('1.26'), Decimal('0.563464')),
 ("Assassin's Creed: Unity", 'ACTION', Decimal('1.25'), Decimal('0.563464')),
 ('The Elder Scrolls V: Skyrim', 'ROLE-PLAYING', Decimal('1.24'), Decimal('0.563464')),
 ("Assassin's Creed Origins", 'ACTION', Decimal('1.22'), Decimal('0.563464')),
 ('Dying Light', 'ACTION', Decimal('1.18'), Decimal('0.563464')),
 ('Far Cry 4', 'SHOOTER', Decimal('1.18'), Decimal('0.563464')),
 ('FIFA 16', 'SPORTS', Decimal('1.15'), Decimal('0.563464')),
 ('Crash Bandicoot N. Sane Trilogy', 'PLATFORM', Decimal('1.09'), Decimal('0.563464')),
 ("Assassin's Creed IV: Black Flag", 'ACTION', Decimal('1.07'), Decimal('0.563464')),
 ("Tom Clancy's Ghost Recon Wildlands", 'SHOOTER', Decimal('1.07'), Decimal('0.563464')),
 ('Dragon Age III: Inquisition', 'ROLE-PLAYING', Decimal('1.07'), Decimal('0.563464')),
 ('Monster Hunter: World', 'ACTION', Decimal('1.03'), Decimal('0.563464')),
 ('Middle-Earth: Shadow of Mordor', 'ACTION', Decimal('1.03'), Decimal('0.563464')),
 ('Dark Souls III

Shooter and Action games dominate the North American Sales, with Sports being the 3rd most popular amongst consumers.
Only three games sold more than 5 million copies. These genre's are most popular and should be prioritized.

TOP 3: Black Ops 3, GTAV, RDR2

# Percentage of Global Sales per Genre #

Count the games in each genre and their total sales compared to the Global Sales. 

GROUP BY, JOIN

In [34]:
%%sql
SELECT 
    COUNT(DISTINCT(Game)) AS GamesInGenre, 
    UCASE(g.GenreName) AS Genre,
    SUM(NASales) AS TotalNASales,
    (SUM(NASales) / SUM(GlobalSales) * 100) AS PercentOfGlobalSales
FROM GameSales s
JOIN Genre g
    ON s.GenreID = g.GenreID
GROUP BY GenreName
ORDER BY GamesInGenre DESC;

 * mysql+mysqldb://admin:***@lmu-dev-01.c205jaszqskd.us-east-1.rds.amazonaws.com/sql_project
   mysql://admin:***@lmu-dev-01.c205jaszqskd.us-east-1.rds.amazonaws.com/sql_project
9 rows affected.


[(129, 'ACTION', Decimal('77.84'), Decimal('34.681875')),
 (50, 'SHOOTER', Decimal('49.85'), Decimal('37.254316')),
 (45, 'SPORTS', Decimal('32.94'), Decimal('35.988201')),
 (43, 'ROLE-PLAYING', Decimal('20.93'), Decimal('35.607349')),
 (32, 'MISC', Decimal('7.81'), Decimal('32.460515')),
 (23, 'RACING', Decimal('4.97'), Decimal('21.157940')),
 (20, 'FIGHTING', Decimal('8.31'), Decimal('44.822006')),
 (15, 'PLATFORM', Decimal('4.97'), Decimal('29.183793')),
 (1, 'STRATEGY', Decimal('0.16'), Decimal('51.612903'))]

Action games has more than double the amount of titles as Shooter games. Sports in 3rd with 5 less titles than Shooter. 45% of Fighting games are sold in North America. For the Fighting genre, I would increase efforts of advertising these games to other regions like Japan and the EU.

# Highest Scoring Games # 

Get the games and critic scores, ranked from High - Low

CASE

In [38]:
%%sql
SELECT 
    DISTINCT(Name),
    CriticScore,
    CASE 
        WHEN CriticScore < 8 THEN 'Low'
        WHEN CriticScore BETWEEN 8 AND 9.2 THEN 'Average'
        WHEN CriticScore BETWEEN 9.3 AND 9.9 THEN 'High'
        ELSE 'N/A'
    END AS ScoreLevel
FROM GameScore
WHERE CriticScore != 0
ORDER BY CriticScore DESC;

 * mysql+mysqldb://admin:***@lmu-dev-01.c205jaszqskd.us-east-1.rds.amazonaws.com/sql_project
   mysql://admin:***@lmu-dev-01.c205jaszqskd.us-east-1.rds.amazonaws.com/sql_project
31 rows affected.


[('Red Dead Redemption 2    ', Decimal('9.8'), 'High'),
 ('God of War (2018)    ', Decimal('9.7'), 'High'),
 ('Grand Theft Auto V    ', Decimal('9.7'), 'High'),
 ('The Last of Us Remastered    ', Decimal('9.5'), 'High'),
 ('Minecraft    ', Decimal('9.4'), 'High'),
 ('Monster Hunter: World    ', Decimal('9.3'), 'High'),
 ("Uncharted 4: A Thief's End    ", Decimal('9.2'), 'Average'),
 ('Battlefield 1    ', Decimal('9.1'), 'Average'),
 ('Horizon: Zero Dawn    ', Decimal('9.1'), 'Average'),
 ("Marvel's Spider-Man    ", Decimal('9.1'), 'Average'),
 ('NBA 2K17    ', Decimal('9.0'), 'Average'),
 ('FIFA 17    ', Decimal('8.9'), 'Average'),
 ('Ghost of Tsushima    ', Decimal('8.8'), 'Average'),
 ('Overwatch    ', Decimal('8.8'), 'Average'),
 ('NBA 2K16    ', Decimal('8.7'), 'Average'),
 ("Assassin's Creed Syndicate    ", Decimal('8.6'), 'Average'),
 ('Fallout 4    ', Decimal('8.6'), 'Average'),
 ('Resident Evil VII: Biohazard    ', Decimal('8.6'), 'Average'),
 ('Call of Duty: Advanced Warfare    ', Decimal('8.5'), 'Average'),
 ('FIFA 16    ', Decimal('8.5'), 'Average'),
 ('Far Cry 4    ', Decimal('8.4'), 'Average'),
 ('Battlefield 4    ', Decimal('8.3'), 'Average'),
 ('FIFA 18    ', Decimal('8.3'), 'Average'),
 ('Batman: Arkham Knight    ', Decimal('8.1'), 'Average'),
 ('Call of Duty: Black Ops 3    ', Decimal('8.1'), 'Average'),
 ('FIFA 15    ', Decimal('8.1'), 'Average'),
 ('Final Fantasy XV    ', Decimal('8.1'), 'Average'),
 ('Call of Duty: WWII    ', Decimal('8.0'), 'Average'),
 ('Far Cry 5    ', Decimal('7.9'), 'Low'),
 ('Call of Duty: Ghosts    ', Decimal('7.5'), 'Low'),
 ('Star Wars Battlefront (2015)    ', Decimal('7.1'), 'Low')]

2 0f 3 top sellers had a high critic score of 9.7 and above(RDR2, GTAV)
There are no Shooter games with a High score.


The #1 seller COD:Black Ops 3 has an average score of 8.1.
COD: Ghosts was the lowest selling game of the Call of Duty franchise and also had the lowest critic score.

Although critic scores do not always line up with the consumers choice, they can be indictive of a good or bad game